core/src/main/java/inetsoft/report/PDFPrinter.java

```diff
@@ -2505,6 +2505,7 @@ private void emitString(String str) {
          }
       }
 
+      boolean hasArabicDigits = str.matches(".*[0-9].*");
```

Comment:
Probably be less complex and avoid this static code analysis if you just do something like:

```java
boolean hasArabicDigits = str.chars().anyMatch(n -> Character.isDigit((char) n));
```